mongorestore/mongorestore_test.go

```diff
@@ -487,3 +487,96 @@ func TestKnownCollections(t *testing.T) {
 		})
 	})
 }
+
+
+func TestFixHashedIndexes(t *testing.T) {
+	testtype.SkipUnlessTestType(t, testtype.IntegrationTestType)
+	session, err := testutil.GetBareSession()
+	if err != nil {
+		t.Fatalf("No server available")
+	}
+
+	type indexRes struct {
+		Key bson.D
+	}
+
+	Convey("Test MongoRestore with hashed indexes and --fixHashedIndexes", t, func() {
+		args := []string{
+			FixDottedHashedIndexesOption,
+		}
+
+		restore, err := getRestoreWithArgs(args...)
+		So(err, ShouldBeNil)
+
+		session, _ = restore.SessionProvider.GetSession()
+		db := session.Database("testdata")
+
+		defer func() {
+			db.Collection("hashedIndexes").Drop(nil)
+		}()
+
+		Convey("Once collection hashedIndexes has been restored, it should exist in restore.knownCollections", func() {
```

Comment:
This description needs to change. restore.knownCollections isn't relevant.
I'd suggest something like: "index {a.b: "hashed"} should be converted to {a.b: 1}"